nf/flows.py

```diff
@@ -175,8 +175,8 @@ def reset_parameters(self):
         init.uniform_(self.initial_param, -math.sqrt(0.5), math.sqrt(0.5))
 
     def forward(self, x):
-        z = torch.zeros_like(x)
-        log_det = torch.zeros(z.shape[0])
+        z = torch.zeros_like(x).to(x.device)
```

Comment:
`torch.zeros_like` should already match the device of its input. 

So no need for `.to(x.device)` on all lines that use `torch.zeros_like`